src/ess/reduce/streaming.py

```diff
@@ -323,6 +357,26 @@ def __init__(
         self._target_keys = target_keys
         self._allow_bypass = allow_bypass
 
+    def set_context(self, context: dict[sciline.typing.Key, Any]) -> None:
+        """
+        Set the context for processing chunks.
+
+        Parameters
+        ----------
+        context:
+            Context to be set.
+        """
+        needs_recompute = set()
+        for key in context:
+            if key not in self._context_keys:
+                raise ValueError(f"Key '{key}' is not a context key")
+            needs_recompute |= self._context_dependencies[key]
+        for key, value in context.items():
+            self._context_workflow[key] = value
+        results = self._context_workflow.compute(needs_recompute)
+        for key, value in results.items():
+            self._process_chunk_workflow[key] = value
```

Comment:
I don't completely follow the graph operations. Does `self._context_dependencies` contain all nodes that depend on the context? Or only those nodes that are direct parents to dynamic or output nodes? If the former, this would cache a lot of data.

Reply:
Only the direct parents, i.e., it computes as far "down" in the graph as possible.

Reply:
I renamed the variable, does that make it clear enough?